initial ansible playbooks/roles

jmazzitelli opened the pull request:
@pilhuhn here's a PR with an initial set of Ansible roles/playbooks. See README.adoc for quick-start docs. I will blog about this, but first wanted to get your take.

Note that the Wildfly roles are really here just to make things easy on the person testing these. It does a download/install of WF 10 (so we can install an agent into it) and start/stop/status of the WildFly instances. The role we really care about is "fetch-and-install-hawkular-wildfly-agent" since that does the download and installation of the agent.

jmazzitelli commented:
Note that each file blows up because of the license header stuff. I'll add those once I find out what a yaml header file looks like :)

jmazzitelli committed:
Initial ansible playbooks/roles.

pilhuhn commented on hawkular-wildfly-agent-ansible-playbook/roles/fetch-and-install-hawkular-wildfly-agent/tasks/main.yml:
```diff
@@ -0,0 +1,63 @@
+  #  validate_certs=no
+
+  shell:
+    "curl --silent --show-error --create-dirs --user '{{hawkular_server_url_username}}:{{hawkular_server_url_password}}' --output '{{hawkular_download_tmpdir}}/hawkular-wildfly-agent-installer.jar' {{hawkular_server_url}}/hawkular/wildfly-agent/installer"
```
Can we guarantee that curl is installed?

jmazzitelli replied:
We could use ansible to install curl :) but really, this was just a stop-gap for Ansible 1.9. Moving to Ansible 2.0 and I would have been able to use the uri: module instead (I wanted to use uri: module now, but I'm on Ansible 1.9 (Fedora 23 doesn't have 2.0 in its dnf repos) and that bug I mentioned in the comments breaks the uri: module).

pilhuhn commented on hawkular-wildfly-agent-ansible-playbook/roles/fetch-and-unpack-wildfly/tasks/main.yml:
```diff
@@ -0,0 +1,27 @@
+    file: path="{{wildfly_install_dir}}" state=directory mode=0755
+
+  - name: Download and unpack WildFly app server
+    unarchive:
```
This always pulls from the remote repo no matter on how many hosts we install.
I had a playbook for pulling that into the ansible "server" and then use that url. See https://github.com/pilhuhn/hawkular-agent-ansible/blob/master/roles/fetch_wildfly/tasks/main.yml

jmazzitelli replied:
This task actually isn't used by my playbooks (the comments I put in here says this only works on Ansible 2.0 and I'm on Ansible 1.9). But I assume your same comment can apply to the task yml file I do use (fetch-wildfly.yml)

theute committed:
Fix directory issue (#195)

The directory needs to be created on the same machine as where WF will be downloaded

jmazzitelli commented:
closing - we don't want to merge this anytime soon.
